types/config.go

```diff
@@ -44,126 +43,147 @@ func GetConfig() *Config {
 		fullFundraiserPath: FullFundraiserPath,
 		txEncoder:          nil,
 	}
-	return sdkConfig
-}
 
-func (config *Config) assertNotSealed() {
-	config.mtx.Lock()
-	defer config.mtx.Unlock()
-
-	if config.sealed {
-		panic("Config is sealed")
-	}
+	return sdkConfig
 }
 
 // SetBech32PrefixForAccount builds the Config with Bech32 addressPrefix and publKeyPrefix for accounts
 // and returns the config instance
 func (config *Config) SetBech32PrefixForAccount(addressPrefix, pubKeyPrefix string) {
-	config.assertNotSealed()
+	config.Lock()
+	defer config.Unlock()
 	config.bech32AddressPrefix["account_addr"] = addressPrefix
 	config.bech32AddressPrefix["account_pub"] = pubKeyPrefix
 }
 
 // SetBech32PrefixForValidator builds the Config with Bech32 addressPrefix and publKeyPrefix for validators
 //  and returns the config instance
 func (config *Config) SetBech32PrefixForValidator(addressPrefix, pubKeyPrefix string) {
-	config.assertNotSealed()
+	config.Lock()
+	defer config.Unlock()
 	config.bech32AddressPrefix["validator_addr"] = addressPrefix
 	config.bech32AddressPrefix["validator_pub"] = pubKeyPrefix
 }
 
 // SetBech32PrefixForConsensusNode builds the Config with Bech32 addressPrefix and publKeyPrefix for consensus nodes
 // and returns the config instance
 func (config *Config) SetBech32PrefixForConsensusNode(addressPrefix, pubKeyPrefix string) {
-	config.assertNotSealed()
+	config.Lock()
+	defer config.Unlock()
 	config.bech32AddressPrefix["consensus_addr"] = addressPrefix
 	config.bech32AddressPrefix["consensus_pub"] = pubKeyPrefix
 }
 
 // SetTxEncoder builds the Config with TxEncoder used to marshal StdTx to bytes
 func (config *Config) SetTxEncoder(encoder TxEncoder) {
-	config.assertNotSealed()
+	config.Lock()
+	defer config.Unlock()
 	config.txEncoder = encoder
 }
 
 // SetAddressVerifier builds the Config with the provided function for verifying that addresses
 // have the correct format
 func (config *Config) SetAddressVerifier(addressVerifier func([]byte) error) {
-	config.assertNotSealed()
+	config.Lock()
+	defer config.Unlock()
 	config.addressVerifier = addressVerifier
 }
 
-// Set the BIP-0044 CoinType code on the config
+// SetCoinType the BIP-0044 CoinType code on the config
 func (config *Config) SetCoinType(coinType uint32) {
-	config.assertNotSealed()
+	config.Lock()
+	defer config.Unlock()
 	config.coinType = coinType
 }
 
-// Set the FullFundraiserPath (BIP44Prefix) on the config
+// SetFullFundraiserPath the FullFundraiserPath (BIP44Prefix) on the config
 func (config *Config) SetFullFundraiserPath(fullFundraiserPath string) {
-	config.assertNotSealed()
+	config.Lock()
+	defer config.Unlock()
 	config.fullFundraiserPath = fullFundraiserPath
 }
 
-// Seal seals the config such that the config state could not be modified further
-func (config *Config) Seal() *Config {
```

Comment:
This is method I was referring to in my previous comment https://github.com/cosmos/cosmos-sdk/pull/5907/files#r401906752